semantic_inference_ros/src/pointcloud_projection.cpp

```diff
@@ -150,7 +190,22 @@ bool projectSemanticImage(const ProjectionConfig& config,
     }
 
     const auto in_view = u >= 0 && u < image.cols && v >= 0 && v < image.rows;
-    *label_iter = in_view ? getter(v, u) : config.unknown_label;
+
+    if (in_view) {
+      *label_out_iter = getter(v, u);
+    } else if (label_in_iter) {
+      int16_t input_label = std::visit(
+          [](auto& iter) { return static_cast<int16_t>(*iter); }, *label_in_iter);
```

Comment:
This should be `int32_t` instead of `int16_t`, otherwise you will lose precision (the projection config should technically also probably use `int32_t` as well, but I copied it from other parts of `semantic_inference` where I have to use `int16_t`)